### PR TITLE
CI/CD Workflows

### PR DESCRIPTION
**Pull request by francardoso93**

### CODE REVIEW

File: .github/workflows/deploy-dev.yml [hunk 1]
```diff
@@ -15,9 +15,9 @@ jobs:
     with:
       docker_image_version_suffix_label: dev
     secrets:
-      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID }}
+      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID_PROD }}
       aws_role_id: ${{ secrets.AWS_ROLE_ID }}
-      aws_region: ${{ secrets.AWS_REGION }}
+      aws_region: us-west-2
```
Comment (gobengo) [inline .github/workflows/deploy-dev.yml:20]: would this be better as a secret/variable?

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -12,7 +12,8 @@ jobs:
     with:
       docker_push: false
     secrets:
-      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID }}
+      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID_PROD }}
       aws_role_id: ${{ secrets.AWS_ROLE_ID }}
-      aws_region: ${{ secrets.AWS_REGION }}
-      aws_ecr_repository: ${{ secrets.AWS_ECR_REPOSITORY }}
+      aws_region: us-west-2
```
Comment (gobengo) [inline .github/workflows/pr.yaml:17]: why not use variable?

File: .github/workflows/deploy-main.yml [hunk 3]
```diff
@@ -48,10 +48,10 @@ jobs: # We need two separate builds as each is going to a different ECR repo
...
       deploy_to_environment: prod
-      lambda: event-delivery
+      lambda: indexer
       docker_image_version: ${{ needs.build-prod.outputs.build_date }}
     secrets:
-      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID }}
+      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID_PROD }}
       aws_role_id: ${{ secrets.AWS_ROLE_ID }}
-      aws_region: ${{ secrets.AWS_REGION }}
+      aws_region: us-west-2
```
Comment (gobengo) [inline .github/workflows/deploy-main.yml:56]: why not use variable?

File: .github/workflows/deploy-main.yml [hunk 4]
```diff
@@ -34,12 +34,12 @@ jobs: # We need two separate builds as each is going to a different ECR repo
...
       deploy_to_environment: staging
-      lambda: staging-ep-event-delivery
+      lambda: staging-ep-indexer
       docker_image_version: ${{ needs.build-staging.outputs.build_date }}
     secrets:
-      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID }}
+      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID_PROD }}
       aws_role_id: ${{ secrets.AWS_ROLE_ID }}
-      aws_region: ${{ secrets.AWS_REGION }}
+      aws_region: us-west-2
```
Comment (gobengo) [inline .github/workflows/deploy-main.yml:42]: why not use variable?

File: .github/workflows/deploy-main.yml [hunk 5]
```diff
@@ -34,12 +34,12 @@ jobs: # We need two separate builds as each is going to a different ECR repo
     needs: build-staging
     with:
       deploy_to_environment: staging
-      lambda: staging-ep-event-delivery
+      lambda: staging-ep-indexer
```
Comment (gobengo) [inline .github/workflows/deploy-main.yml:37]: why mention 'indexer' in here if this is no longer the indexer lambda?

File: .github/workflows/deploy-main.yml [hunk 6]
```diff
@@ -13,19 +13,19 @@ jobs: # We need two separate builds as each is going to a different ECR repo
...
     permissions:
       packages: write
       id-token: write
     uses: elastic-ipfs/shared-workflows/.github/workflows/lambda-shared-build.yaml@main
     secrets:
-      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID }}
+      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID_PROD }}
       aws_role_id: ${{ secrets.AWS_ROLE_ID }}
-      aws_region: ${{ secrets.AWS_REGION }}
+      aws_region: us-west-2
```
Comment (gobengo) [inline .github/workflows/deploy-main.yml:28]: should we use variable?

File: .github/workflows/deploy-main.yml [hunk 7]
```diff
@@ -13,19 +13,19 @@ jobs: # We need two separate builds as each is going to a different ECR repo
...
```
Comment (gobengo) [inline .github/workflows/deploy-main.yml:18]: should we use variable?

File: .github/workflows/deploy-dev.yml [hunk 8]
```diff
@@ -26,10 +26,10 @@ jobs:
...
       deploy_to_environment: dev
-      lambda: dev-ep-event-delivery
+      lambda: dev-ep-indexer
       docker_image_version: ${{ needs.build.outputs.build_date }}-dev
     secrets:
-      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID }}
+      aws_account_id: ${{ secrets.AWS_ACCOUNT_ID_PROD }}
       aws_role_id: ${{ secrets.AWS_ROLE_ID }}
-      aws_region: ${{ secrets.AWS_REGION }}
+      aws_region: us-west-2
```
Comment (gobengo) [inline .github/workflows/deploy-dev.yml:34]: should we use variable?

File: .github/workflows/deploy-dev.yml [hunk 9]
```diff
@@ -26,10 +26,10 @@ jobs:
     needs: build
     with:
       deploy_to_environment: dev
-      lambda: dev-ep-event-delivery
+      lambda: dev-ep-indexer
```
Comment (gobengo) [inline .github/workflows/deploy-dev.yml:29]: why mention 'indexer' in here if this is no longer the indexer lambda?

File: .github/workflows/deploy-main.yml (hunk 5 above)
Comment (francardoso93) [reply .github/workflows/deploy-main.yml]: Fixed

File: .github/workflows/deploy-main.yml [hunk 10]
```diff
@@ -48,10 +48,10 @@ jobs: # We need two separate builds as each is going to a different ECR repo
     needs: build-prod
     with:
       deploy_to_environment: prod
-      lambda: event-delivery
+      lambda: indexer
```
[the remark this answers is not in the document]
Comment (francardoso93) [reply .github/workflows/deploy-main.yml]: Fixed

File: .github/workflows/pr.yaml [hunk 11]
```diff
@@ -12,7 +12,8 @@ jobs:
...
+      aws_ecr_repository: whatever
```
[the remark this answers is not in the document]
Comment (francardoso93) [reply .github/workflows/pr.yaml:18]: PR build won't push to ECR repo

File: .github/workflows/deploy-dev.yml (hunk 1 above)
Comment (francardoso93) [reply .github/workflows/deploy-dev.yml:20]: Yes! But we are using a shared template. Need to keep this like that for now for compatibility purposes. Need to change all repos that use this template before doing that.

File: .github/workflows/pr.yaml (hunk 2 above)
Comment (francardoso93) [reply .github/workflows/pr.yaml:17]: https://github.com/elastic-ipfs/event-delivery-lambda/pull/1#discussion_r930239640

File: .github/workflows/deploy-main.yml (hunk 3 above)
Comment (francardoso93) [reply .github/workflows/deploy-main.yml:56]: https://github.com/elastic-ipfs/event-delivery-lambda/pull/1#discussion_r930239640

File: .github/workflows/deploy-main.yml (hunk 4 above)
Comment (francardoso93) [reply .github/workflows/deploy-main.yml:42]: https://github.com/elastic-ipfs/event-delivery-lambda/pull/1#discussion_r930239640

File: .github/workflows/deploy-main.yml (hunk 7 above)
Comment (francardoso93) [reply .github/workflows/deploy-main.yml:18]: https://github.com/elastic-ipfs/event-delivery-lambda/pull/1#discussion_r930239640

File: .github/workflows/deploy-dev.yml (hunk 8 above)
Comment (francardoso93) [reply .github/workflows/deploy-dev.yml:34]: https://github.com/elastic-ipfs/event-delivery-lambda/pull/1#discussion_r930239640

File: .github/workflows/deploy-main.yml (hunk 6 above)
Comment (francardoso93) [reply .github/workflows/deploy-main.yml:28]: https://github.com/elastic-ipfs/event-delivery-lambda/pull/1#discussion_r930239640

**Pull request merged**